testsuite/framework/src/main/java/org/keycloak/testsuite/client/Keycloak.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.testsuite.client;
+
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.WebTarget;
+import org.keycloak.admin.client.Config;
+import org.keycloak.admin.client.resource.BearerAuthFilter;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.admin.client.resource.RealmsResource;
+import org.keycloak.admin.client.resource.ServerInfoResource;
+import org.keycloak.admin.client.spi.ResteasyClientProvider;
+import org.keycloak.admin.client.token.TokenManager;
+
+import javax.net.ssl.SSLContext;
+import java.net.URI;
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+import static org.keycloak.OAuth2Constants.PASSWORD;
+
+/**
+ * Provides a Keycloak client. By default, this implementation uses a the default RestEasy client builder settings.
+ * To customize the underling client, use a {@link org.keycloak.admin.client.KeycloakBuilder} to create a Keycloak client.
+ *
+ * To read Responses, you can use {@link org.keycloak.admin.client.CreatedResponseUtil} for objects created
+ *
+ * @author rodrigo.sasaki@icarros.com.br
+ * @see org.keycloak.admin.client.KeycloakBuilder
+ */
+public class Keycloak implements AutoCloseable {
```

Comment:
Why we need to copy this class in the testsuite?

IMO in the testsuite we should test the same admin-client, which is used by community/customers.  So we should not need any additional `Keycloak` class, but should use the one from the module `admin-client` IMO.

If you think that there is some limitation in the existing `Keycloak` class, then the possible way to go is to update the `Keycloak` class in the Keycloak server codebase, send the PR to keycloak main and then as a follow-up, sync the `Keycloak` class to keycloak-client into this PR to make sure that it contains all the necessary stuff. WDYT?

Reply:
Copied Keycloak into the test suite only because of difficulties with the SSL context when creating a client with client credentials grant but found a way to do it by removing Keycloak from the test suite and without adding anything to the original Keycloak